From b2471098db7851fc1ccfc3cbeed4e64a7ae37ecd Mon Sep 17 00:00:00 2001
From: Pavithra R <quic_pavir@quicinc.com>
Date: Fri, 3 Nov 2023 22:43:32 +0530
Subject: [PATCH 365/500] net: Fix the crash in skbuff.c for debug build

When skb is allocated in NAPI context we aren't
initializing and activating the skb object. Add support
for init and activating skb in napi_build_skb.

Change-Id: Idcc3d6c260831be12ec85df474bcdc17f86d3292
Signed-off-by: Pavithra R <quic_pavir@quicinc.com>
---
 net/core/skbuff.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index 786b5bdd205e..b471121e44c9 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -541,6 +541,8 @@ static struct sk_buff *__napi_build_skb(void *data, unsigned int frag_size)
 	if (unlikely(!skb))
 		return NULL;
 
+	skbuff_debugobj_init_and_activate(skb);
+
 	memset(skb, 0, offsetof(struct sk_buff, tail));
 	__build_skb_around(skb, data, frag_size);
 
-- 
2.34.1

